pkg/proxy/roundrobin.go

```diff
@@ -210,14 +236,14 @@ func (lb *LoadBalancerRR) OnUpdate(endpoints []api.Endpoints) {
 	for _, endpoint := range endpoints {
 		existingEndpoints, exists := lb.endpointsMap[endpoint.Name]
 		validEndpoints := filterValidEndpoints(endpoint.Endpoints)
-		if !exists || !reflect.DeepEqual(existingEndpoints, validEndpoints) {
+		if !exists || !endpointsAreEqual(existingEndpoints, validEndpoints) {
```

Comment:
Just document it.  The fact that it returns a reference to its input is what makes it useful in functional-style ways.  It's a well-established pattern in lots of languages - just document that it does an in-place sort.  Or maybe call it SortStringsInPlace() if it makes you more comfortable